### PR TITLE
CON-107 Implement super_admin functionality

### PR DESCRIPTION
**Pull request by vic3king**
## Description ##
This pull request introduces the 'Super_Admin' role to our application. Currently, there is no way for a user to be able to view analytics for any of our supported locations. this PR makes that possible and also makes it possible for the 'Super_Admin' to perform the same tasks that a normal 'Admin' would be able to perform.

## How can This be tested ##
- Clone the repo: `git clone https://github.com/andela/mrm_api.git`
- Setup project according to `Readme.md`
- checkout to branch `story/CON-107-create-enable-super-admin`
- on insomnia
create a new 'Super_Admin' role by running this mutation
```
mutation {
  createRole(role:"Super_Admin"){
    role {
      role
    }
  }
}
```
- next upgrade your user role to the new 'Super_Admin' role.
```
mutation {
  changeUserRole(email: "youremail@andela.com", roleId: 3) {
    user {
      email
      roles {
        id
      }
    }
  }
}
```
- next up, try getting analytics for any location of choice. e.g(locationId:1 || 2 || 2)
```
    query {
      allAnalytics(startDate:"jul 11 2018", endDate:"jul 12 2018", locationId:2) {
          checkinsPercentage
          appBookingsPercentage
          autoCancellationsPercentage
          cancellationsPercentage
          bookings
          analytics{
            roomName
            cancellations
            cancellationsPercentage
            autoCancellations
            numberOfBookings
            checkins
            checkinsPercentage
            bookingsPercentageShare
            appBookings
            appBookingsPercentage
            events{
              durationInMinutes
              }
          }
          bookingsCount{
            totalBookings
            period
          }
        }
  }
```
- you can also try performing any admin specific duty using the 'Super_Admin' role and see it work

## Type of change ##
Please select the relevant option
- [ ] Bug fix(a non-breaking change which fixes an issue)
- [x] New feature(a non-breaking change which adds functionality)
- [ ] Breaking change(fix of a feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist: ##
- [x] My code follows the style guidelines of this project
- [x] I have linted my code prior to submission
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

## JIRA ##
[CON-107](https://andela-apprenticeship.atlassian.net/browse/CON-107)

## Relevant screenshots: ##
N/A

### CODE REVIEW

**Comment by vic3king** [comment]
> Nice implementation @vic3king. The application works as we would want it to
> I suggest that you also verify if a location exists, for example, I tried to query for `locationId:10` but the query just went through

Hey @mnswaleh thanks for the feedback. I have implemented and tested it out. Please confirm.

File: api/analytics/all_analytics_query.py [hunk 1]
```diff
@@ -41,22 +43,30 @@ class Query(graphene.ObjectType):
...
+    @Auth.user_roles('Admin', 'Default User', 'Super_Admin')
     def resolve_all_analytics(self, info, **kwargs):
         start_date = kwargs.get('start_date')
         end_date = kwargs.get('end_date')
+        location_id = admin_roles.user_location_for_analytics_view()
+
+        admin_details = get_user_from_db()
+        # check that id is valid
+        verify_location_id(kwargs)
+        if admin_details.roles[0].id == 3 and kwargs.get('location_id', None):
```
Comment (joshuaocero) [inline api/analytics/all_analytics_query.py]: Please avoid hardcoding IDs in the codebase

File: api/analytics/all_analytics_query.py [hunk 2]
```diff
@@ -41,22 +43,30 @@ class Query(graphene.ObjectType):
...
-            'end': end_date
+            'end': end_date,
         }
-        location_id = admin_roles.user_location_for_analytics_view()
         start_date, end_date = CommonAnalytics.all_analytics_date_validation(
             self, start_date, end_date
         )
         query = Room.get_query(info)
-        room_analytics, bookings, percentages_dict, bookings_count = AllAnalyticsHelper.get_all_analytics( # noqa
+        room_analytics, bookings, percentages_dict, bookings_count = AllAnalyticsHelper.get_all_analytics(  # noqa
```
Comment (joshuaocero) [inline api/analytics/all_analytics_query.py]: This change is unnecessary

**Comment by vic3king** [comment]
> @vic3king great start but this task is not complete.
> 
> You seem to have handled only part one of this task ("A super-admin will be able to view analytics across all platforms."). You have not handled the second part ("They can perform all web app operations across all platforms.").
> 
> There are also a few issues I have pointed out in this review. Please look into them.

@joshuaocero Thanks for the feedback. On our application, our `user_role` function handles authentication and verifies that a user is of a certain type. when the function is called it receives a couple of arguments which are the names of diffrent user roles. all I had to do to complete the second part of this task was to create the new user role and then pass that role name as a parameter to any mutation or query where we called the user_role function. `@Auth.user_roles('Admin', 'Super_Admin')`

File: api/analytics/all_analytics_query.py (hunk 1 above)
Comment (vic3king) [reply api/analytics/all_analytics_query.py]: @joshuaocero I'll refactor this and come up with a better implementation.

**Pull request merged**